python/pyspark/sql/context.py

```diff
@@ -72,8 +72,13 @@ def __init__(self, sparkContext, sparkSession=None, jsqlContext=None):
         self._sc = sparkContext
         self._jsc = self._sc._jsc
         self._jvm = self._sc._jvm
+
         if sparkSession is None:
-            sparkSession = SparkSession(sparkContext)
+            if sparkContext is SparkContext._active_spark_context:
+                sparkSession = SparkSession.builder.getOrCreate()
+            else:
```

Comment:
when will hit the else branch? The if-else seems the only difference between this PR and https://github.com/apache/spark/pull/16369